sdks/java/io/snowflake/src/test/java/org/apache/beam/sdk/io/snowflake/test/SnowflakeIOIT.java

```diff
@@ -0,0 +1,174 @@
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.snowflake.test;
+
+import static org.apache.beam.sdk.io.common.IOITHelper.readIOTestPipelineOptions;
+import static org.apache.beam.sdk.io.snowflake.test.TestUtils.SnowflakeIOITPipelineOptions;
+import static org.apache.beam.sdk.io.snowflake.test.TestUtils.getTestRowCsvMapper;
+import static org.apache.beam.sdk.io.snowflake.test.TestUtils.getTestRowDataMapper;
+
+import java.sql.SQLException;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.apache.beam.sdk.PipelineResult;
+import org.apache.beam.sdk.coders.SerializableCoder;
+import org.apache.beam.sdk.io.FileSystems;
+import org.apache.beam.sdk.io.GenerateSequence;
+import org.apache.beam.sdk.io.common.HashingFn;
+import org.apache.beam.sdk.io.common.TestRow;
+import org.apache.beam.sdk.io.fs.MatchResult;
+import org.apache.beam.sdk.io.fs.MoveOptions;
+import org.apache.beam.sdk.io.fs.ResourceId;
+import org.apache.beam.sdk.io.snowflake.SnowflakeIO;
+import org.apache.beam.sdk.io.snowflake.credentials.SnowflakeCredentialsFactory;
+import org.apache.beam.sdk.io.snowflake.enums.WriteDisposition;
+import org.apache.beam.sdk.testing.PAssert;
+import org.apache.beam.sdk.testing.TestPipeline;
+import org.apache.beam.sdk.transforms.Combine;
+import org.apache.beam.sdk.transforms.Count;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.values.PCollection;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+
+/**
+ * A test of {@link org.apache.beam.sdk.io.snowflake.SnowflakeIO} on an independent Snowflake
+ * instance.
+ *
+ * <p>This test requires a running instance of Snowflake, configured for your GCP account. Pass in
+ * connection information using PipelineOptions:
+ *
+ * <pre>
+ * ./gradlew -p sdks/java/io/snowflake integrationTest -DintegrationTestPipelineOptions='[
+ * "--serverName=<YOUR SNOWFLAKE SERVER NAME>",
+ * "--username=<USERNAME>",
+ * "--password=<PASSWORD>",
+ * "--database=<DATABASE NAME>",
+ * "--schema=<SCHEMA NAME>",
+ * "--stagingBucketName=gs://<GCS BUCKET NAME>",
+ * "--storageIntegrationName=<STORAGE INTEGRATION NAME>",
+ * "--numberOfRecords=<1000, 100000, 600000, 5000000>",
+ * "--runner=DataflowRunner",
+ * "--region=<GCP REGION FOR DATAFLOW RUNNER>",
+ * "--project=<GCP_PROJECT>"]'
+ * --tests org.apache.beam.sdk.io.snowflake.test.SnowflakeIOIT
+ * -DintegrationTestRunner=dataflow
+ * </pre>
+ */
+public class SnowflakeIOIT {
```

Comment:
What do you think about renaming test into BatchSnowflakeIOIT?

Reply:
I was thinking to do it later as currently there's only batch in beam as far as SnowflakeIO is concerned.
But given the streaming is soon to be PR'd let's do it!